src/openrct2/rct1/S4Importer.cpp

```diff
@@ -1973,26 +1973,62 @@ class S4Importer final : public IParkImporter
     {
         gMapBaseZ = 7;
 
-        for (uint32_t index = 0, dstOffset = 0; index < RCT1_MAX_TILE_ELEMENTS; index++)
+        // Build tile pointer cache (needed to get the first element at a certain location)
+        constexpr const uint16_t MaxTileElementPointers = RCT1_MAX_MAP_SIZE * RCT1_MAX_MAP_SIZE;
+        RCT12TileElement* tilePointers[MaxTileElementPointers];
```

Comment:
You should break this up into a few functions save this function from getting crazy big. One for setup, one for access. That sort of thing.